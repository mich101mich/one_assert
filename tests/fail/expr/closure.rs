fn main() {
    one_assert::assert!(|| true);
    one_assert::assert!(|| 5);
    one_assert::assert!(|x: usize| x + 5);
}
