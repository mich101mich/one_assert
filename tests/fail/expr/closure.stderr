error[E0308]: mismatched types
 --> tests/fail/expr/closure.rs:2:25
  |
2 |     one_assert::assert!(|| true);
  |                         ^^^^^^^ expected `bool`, found closure
  |
  = note: expected type `bool`
          found closure `{closure@$DIR/tests/fail/expr/closure.rs:2:25: 2:27}`
help: use parentheses to call this closure
  |
2 |     one_assert::assert!((|| true)());
  |                         +       +++

error[E0308]: mismatched types
 --> tests/fail/expr/closure.rs:3:25
  |
3 |     one_assert::assert!(|| 5);
  |                         ^^^^ expected `bool`, found closure
  |
  = note: expected type `bool`
          found closure `{closure@$DIR/tests/fail/expr/closure.rs:3:25: 3:27}`

error[E0308]: mismatched types
 --> tests/fail/expr/closure.rs:4:25
  |
4 |     one_assert::assert!(|x: usize| x + 5);
  |                         ^^^^^^^^^^^^^^^^ expected `bool`, found closure
  |
  = note: expected type `bool`
          found closure `{closure@$DIR/tests/fail/expr/closure.rs:4:25: 4:35}`
